ecl/hqlcpp/hqlhtcpp.cpp

```diff
@@ -2351,6 +2351,21 @@ void ActivityInstance::buildPrefix()
 
 void ActivityInstance::buildSuffix()
 {
+    if (!implementationClassName && constructorArgs)
+    {
+        StringBuffer baseClassName;
+        baseClassName.append("CThor").append(activityArgName).append("Arg");
```

Comment:
Is this calling the constructor mentioned in comments above? If so not sure how it would work unless activityArgName was "External".

Reply:
It would only work if an entry was added for any other based class that required it.